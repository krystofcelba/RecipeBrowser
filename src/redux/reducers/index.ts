import { combineReducers } from 'redux';
import {
  Actions as NavActions,
  reducer as navReducer,
  State as NavState,
} from './nav';
import {
  Actions as RecipesActions,
  reducer as recipesReducer,
  State as RecipesState,
} from './recipes';

export type RootAction =
  | NavActions[keyof NavActions]
  | RecipesActions[keyof RecipesActions];

interface StoreEnhancerState {}

export interface RootState extends StoreEnhancerState {
  nav: NavState;
  recipes: RecipesState;
}

export default combineReducers<RootState>({
  nav: navReducer,
  recipes: recipesReducer,
});
