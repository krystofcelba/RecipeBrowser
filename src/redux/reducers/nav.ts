export const ON_NAV_SCREEN_CHANGED = 'ON_NAV_SCREEN_CHANGE';

export type Actions = {
  ON_NAV_SCREEN_CHANGE: {
    type: typeof ON_NAV_SCREEN_CHANGED;
    screenId: string;
  };
};

export const actionCreators = {
  onNavScreenChange: (
    screenId: string,
  ): Actions[typeof ON_NAV_SCREEN_CHANGED] => ({
    type: ON_NAV_SCREEN_CHANGED,
    screenId,
  }),
};

export type State = {
  readonly currentScreenId: string;
};

export const reducer = (
  state = { currentScreenId: '' },
  action: Actions[keyof Actions],
) => {
  switch (action.type) {
    case ON_NAV_SCREEN_CHANGED:
      return { ...state, currentScreenId: action.screenId };
    default:
      return state;
  }
};
