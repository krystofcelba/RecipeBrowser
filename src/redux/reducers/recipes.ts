export const FETCH_RECIPES = 'FETCH_RECIPES';
export const FETCH_RECIPES_SUCCESS = 'FETCH_RECIPES_SUCCESS';
export const FETCH_RECIPES_FAILED = 'FETCH_RECIPES_FAILED';

export type Actions = {
  FETCH_RECIPES: { type: typeof FETCH_RECIPES };
  FETCH_RECIPES_SUCCESS: { type: typeof FETCH_RECIPES_SUCCESS; recipes: [{}] };
  FETCH_RECIPES_FAILED: { type: typeof FETCH_RECIPES_FAILED };
};

export const actionCreators = {
  fetchRecipes: (): Actions[typeof FETCH_RECIPES] => ({
    type: FETCH_RECIPES,
  }),
  fetchRecipesSuccess: (
    recipes: [{}],
  ): Actions[typeof FETCH_RECIPES_SUCCESS] => ({
    type: FETCH_RECIPES_SUCCESS,
    recipes,
  }),
  fetchRecipesFailed: (): Actions[typeof FETCH_RECIPES_FAILED] => ({
    type: FETCH_RECIPES_FAILED,
  }),
};

export type State = {
  readonly recipes: {};
  readonly loading: boolean;
};

export const reducer = (
  state = { recipes: {}, loading: false },
  action: Actions[keyof Actions],
) => {
  switch (action.type) {
    case FETCH_RECIPES:
      return { ...state, loading: true };
    default:
      return state;
  }
};
