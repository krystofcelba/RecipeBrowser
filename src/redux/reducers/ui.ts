import { Recipe } from 'src/services/api';
export const UPDATE_RECIPES_SEARCH_INPUT_TEXT = 'UPDATE_RECIPES_SEARCH_INPUT_TEXT';

export const SUBMIT_ADD_RECIPE_FORM = 'SUBMIT_ADD_RECIPE_FORM';

export type Actions = {
  UPDATE_RECIPES_SEARCH_INPUT_TEXT: {
    type: typeof UPDATE_RECIPES_SEARCH_INPUT_TEXT;
    text: string;
  };
  SUBMIT_ADD_RECIPE_FORM: {
    type: typeof SUBMIT_ADD_RECIPE_FORM;
    recipe: Recipe;
  };
};

export const actionCreators = {
  updateRecipesSearchInputText: (text: string): Actions[typeof UPDATE_RECIPES_SEARCH_INPUT_TEXT] => ({
    type: UPDATE_RECIPES_SEARCH_INPUT_TEXT,
    text,
  }),
  submitAddRecipeForm: (recipe: Recipe): Actions[typeof SUBMIT_ADD_RECIPE_FORM] => ({
    type: SUBMIT_ADD_RECIPE_FORM,
    recipe,
  }),
};

export type State = {
  readonly recipesList: { searchInput: { text: string } };
};

export const reducer = (
  state = {
    recipesList: { searchInput: { text: '' } },
  },
  action: Actions[keyof Actions],
) => {
  switch (action.type) {
    case UPDATE_RECIPES_SEARCH_INPUT_TEXT: {
      return { ...state, recipesList: { ...state.recipesList, searchInput: { text: action.text } } };
    }
    default:
      return state;
  }
};
