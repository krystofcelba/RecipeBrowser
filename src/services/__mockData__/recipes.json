[
  {
    "description":
      "This buffet classic can also be healthy! Mayonnaise is replaced by cottage cheese. You can blend the filling for more creaminess.",
    "id": 9,
    "image": "https://nutrition-assets.freeletics.com/recipes/large/mustard_deviled_eggs.jpg",
    "ingredients": [
      {
        "amount": 4,
        "name": "hard-boiled eggs",
        "unit": ""
      },
      {
        "amount": 2,
        "name": "Dijon mustard",
        "unit": "tsp"
      },
      {
        "amount": 100,
        "name": "cottage cheese",
        "unit": "g"
      },
      {
        "amount": 2,
        "name": "chopped dill",
        "unit": "tbsp"
      }
    ],
    "name": "Mustard Deviled Eggs",
    "seasonings": ["Paprika powder"],
    "steps": [
      "Halve eggs and remove yolk.",
      "Mix yolks with mustard, cottage cheese and half of dill.",
      "Fill egg halves with yolk mixture and garnish with remaining dill."
    ]
  }
]
