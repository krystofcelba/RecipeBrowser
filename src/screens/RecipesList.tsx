import React, { Component } from 'react';
import { KeyboardAvoidingView, StyleSheet, FlatList, Platform } from 'react-native';
import { connect } from 'react-redux';
import ActionButton from 'react-native-action-button';

import { actionCreators } from '../redux/reducers/recipes';
import { Recipe } from '../services/api';
import { navigatorStyle, navigatorConfig, StylesConstants, Colors } from '../assets/constants';
import RecipeCard from '../components/recipes-list/RecipeCard';
import { getFilteredRecipes } from '../redux/selectors';
import i18n from '../assets/i18n';

interface Props {
  navigator: any;
  fetchRecipes: typeof actionCreators.fetchRecipes;
  recipes: [Recipe];
}

class RecipesList extends Component<Props, any> {
  componentDidMount() {
    this.props.navigator.setStyle({
      ...navigatorStyle,
      navBarCustomView: 'recipeBrowser.RecipesListTopBar',
      navBarComponentAlignment: 'center',
      navBarCustomViewInitialProps: {},
    });
    this.props.fetchRecipes();
  }

  onPressRecipe = (recipe: Recipe) => {
    this.props.navigator.push({
      ...navigatorConfig,
      screen: 'recipeBrowser.RecipeDetail',
      title: recipe.name,
      passProps: { recipeId: recipe.id },
    });
  };

  onPressAddButton = () => {
    this.props.navigator.showModal({
      ...navigatorConfig,
      screen: 'recipeBrowser.AddRecipeForm',
      title: i18n.t('addRecipe'),
    });
  };

  recipeKeyExtractor = item => item.id;

  renderRecipeFlatListItem = ({ item }) => (
    <RecipeCard recipe={item} onPress={this.onPressRecipe} testID={`recipeCard_${item.id}`} />
  );

  render() {
    const { recipes } = this.props;
    return (
      <KeyboardAvoidingView
        testID="recipesListScreen"
        keyboardVerticalOffset={60}
        behavior={Platform.OS === 'ios' ? 'padding' : undefined}
        style={styles.container}
      >
        <FlatList data={recipes} keyExtractor={this.recipeKeyExtractor} renderItem={this.renderRecipeFlatListItem} />
        <ActionButton
          testID="addButton"
          buttonColor={Colors.secondaryColor}
          onPress={this.onPressAddButton}
          hideShadow
        />
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginBottom: StylesConstants.defaultMargin * 1.5,
  },
});

const mapStateToProps = state => ({ recipes: getFilteredRecipes(state) });

const mapDispatchToProps = {
  fetchRecipes: actionCreators.fetchRecipes,
};

export default connect(mapStateToProps, mapDispatchToProps)(RecipesList);
