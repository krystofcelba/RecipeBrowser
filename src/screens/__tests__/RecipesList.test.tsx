import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import sinon from 'sinon';
import { Platform } from 'react-native';

import RecipesList from '../RecipesList';
import { concatIngredientsNames } from '../../redux/reducers/recipes';

const mockedRecipes = require('../../services/__mockData__/recipes.json');

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  recipes: {
    recipesById: mockedRecipes.map(recipe => recipe.id) as [number],
    recipes: mockedRecipes.reduce(
      (recipesHash, recipe) => ({
        ...recipesHash,
        [recipe.id]: {
          ...recipe,
          ingredientsNames: concatIngredientsNames(recipe),
        },
      }),
      {},
    ),
  },
  ui: {
    recipesList: { searchInput: { text: '' } },
  },
};

describe('RecipesList component', () => {
  const setStyleSpy = sinon.spy();
  const showModalSpy = sinon.spy();
  const navigator = {
    setStyle: setStyleSpy,
    showModal: showModalSpy,
  };

  it('renders as expected', () => {
    const wrapper = shallow(<RecipesList navigator={navigator} />, { context: { store: mockStore(initialState) } });
    expect(wrapper.dive()).toMatchSnapshot();
    expect(setStyleSpy.args).toMatchSnapshot();
  });

  it('renders as expected on android', () => {
    Platform.OS = 'android';
    const wrapper = shallow(<RecipesList navigator={navigator} />, { context: { store: mockStore(initialState) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('show modal as expected on add button press', () => {
    const wrapper = shallow(<RecipesList navigator={navigator} />, { context: { store: mockStore(initialState) } });
    const render = wrapper.dive();
    render.find('ActionButton').forEach(child => {
      child.simulate('press');
      expect(showModalSpy.args).toMatchSnapshot();
    });
  });
});
