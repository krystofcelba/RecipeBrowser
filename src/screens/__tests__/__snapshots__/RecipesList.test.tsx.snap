// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesList component call show modal as expected on add button press 1`] = `
Array [
  Array [
    Object {
      "backButtonTitle": "",
      "navigatorStyle": Object {
        "navBarBackgroundColor": "#81b640",
        "navBarButtonColor": "#ffff",
        "navBarTextColor": "#ffff",
        "navBarTextFontFamily": "Avenir",
        "statusBarColor": "#508607",
        "statusBarTextColorScheme": "light",
      },
      "screen": "recipeBrowser.AddRecipeForm",
      "title": "Add Recipe",
    },
  ],
]
`;

exports[`RecipesList component calls push screen as expected on recipe flat list item press 1`] = `
Array [
  Array [
    Object {
      "backButtonTitle": "",
      "navigatorStyle": Object {
        "navBarBackgroundColor": "#81b640",
        "navBarButtonColor": "#ffff",
        "navBarTextColor": "#ffff",
        "navBarTextFontFamily": "Avenir",
        "statusBarColor": "#508607",
        "statusBarTextColorScheme": "light",
      },
      "passProps": Object {
        "recipeId": 9,
      },
      "screen": "recipeBrowser.RecipeDetail",
      "title": "Mustard Deviled Eggs",
    },
  ],
]
`;

exports[`RecipesList component renders as expected 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={60}
  style={
    Object {
      "flex": 1,
    }
  }
  testID="recipesListScreen"
>
  <FlatList
    contentContainerStyle={
      Object {
        "paddingBottom": 10,
      }
    }
    data={
      Array [
        Object {
          "description": "This buffet classic can also be healthy! Mayonnaise is replaced by cottage cheese. You can blend the filling for more creaminess.",
          "id": 9,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/mustard_deviled_eggs.jpg",
          "ingredients": Array [
            Object {
              "amount": 4,
              "name": "hard-boiled eggs",
              "unit": "",
            },
            Object {
              "amount": 2,
              "name": "Dijon mustard",
              "unit": "tsp",
            },
            Object {
              "amount": 100,
              "name": "cottage cheese",
              "unit": "g",
            },
            Object {
              "amount": 2,
              "name": "chopped dill",
              "unit": "tbsp",
            },
          ],
          "ingredientsNames": "hard-boiled eggs;dijon mustard;cottage cheese;chopped dill;",
          "name": "Mustard Deviled Eggs",
          "seasonings": Array [
            "Paprika powder",
          ],
          "steps": Array [
            "Halve eggs and remove yolk.",
            "Mix yolks with mustard, cottage cheese and half of dill.",
            "Fill egg halves with yolk mixture and garnish with remaining dill.",
          ],
        },
        Object {
          "description": "Scrambled eggs are always a great idea. This version with spinach, tomatoes and salty feta is suitable at any point during the day.",
          "id": 2,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/feta_scrambled_eggs.jpg",
          "ingredients": Array [
            Object {
              "amount": 3,
              "name": "eggs",
              "unit": "",
            },
            Object {
              "amount": 100,
              "name": "low-fat feta cheese",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "diced tomato",
              "unit": "",
            },
            Object {
              "amount": 40,
              "name": "fresh spinach",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "coconut oil",
              "unit": "tsp",
            },
          ],
          "ingredientsNames": "eggs;low-fat feta cheese;diced tomato;fresh spinach;coconut oil;",
          "name": "Feta Scrambled Eggs",
          "seasonings": Array [
            "Grated nutmeg",
            "unrefined salt",
            "black pepper",
          ],
          "steps": Array [
            "Heat up coconut oil in a pan. Cook tomatoes and spinach on medium heat for 2 minutes or until spinach softens.",
            "Crack eggs into pan and cook for approx. 4 minutes or desired consistency.",
            "Add grated nutmeg, salt and pepper.",
            "Crumble feta cheese over warm eggs.",
          ],
        },
        Object {
          "description": "This fruity combination of grapefruit, jalapeños and avocado will invite memories of holidays. A side of quickly fried fish and red cabbage.",
          "id": 31,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/fried_fish_with_grapefruit_salsa.jpg",
          "ingredients": Array [
            Object {
              "amount": 200,
              "name": "white fish fillet (cod, plaice, sole; fresh or frozen)",
              "unit": "g",
            },
            Object {
              "amount": 0.5,
              "name": "avocados",
              "unit": "",
            },
            Object {
              "amount": 0.5,
              "name": "grapefruits",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "chopped jalapeño",
              "unit": "",
            },
            Object {
              "amount": 0.5,
              "name": "coriander",
              "unit": "bunches",
            },
            Object {
              "amount": 0.5,
              "name": "limes (juice)",
              "unit": "",
            },
            Object {
              "amount": 100,
              "name": "chopped cabbage",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "coconut oil",
              "unit": "tsp",
            },
          ],
          "ingredientsNames": "white fish fillet (cod, plaice, sole; fresh or frozen);avocados;grapefruits;chopped jalapeño;coriander;limes (juice);chopped cabbage;coconut oil;",
          "name": "Fried Fish With Grapefruit Salsa",
          "seasonings": Array [
            "Chili flakes",
            "unrefined salt",
            "black pepper",
          ],
          "steps": Array [
            "Peel and cube avocado and grapefruit.",
            "In a bowl, mix avocado, grapefruit, jalapeño, coriander and lime juice. Add a pinch of salt.",
            "Heat up coconut oil in a pan. Fry each side of fish fillets for 3 minutes on medium heat.",
            "Serve fried fish on a bed of chopped red cabbage and grapefruit salsa.",
          ],
        },
        Object {
          "description": "A timeless dish, that has a lot to offer beyond its taste! Why not make this restaurant classic on your own?",
          "id": 61,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/classic_tuna_salad.jpg",
          "ingredients": Array [
            Object {
              "amount": 150,
              "name": "water-packed tuna",
              "unit": "g",
            },
            Object {
              "amount": 2,
              "name": "hard-boiled eggs",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "sliced onion",
              "unit": "",
            },
            Object {
              "amount": 50,
              "name": "pitted green olives",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "diced tomato",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "small romaine lettuce",
              "unit": "",
            },
            Object {
              "amount": 0.5,
              "name": "lemons (juice)",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "virgin olive oil",
              "unit": "tsp",
            },
            Object {
              "amount": 1,
              "name": "mixed Italian herbs",
              "unit": "tsp",
            },
          ],
          "ingredientsNames": "water-packed tuna;hard-boiled eggs;sliced onion;pitted green olives;diced tomato;small romaine lettuce;lemons (juice);virgin olive oil;mixed italian herbs;",
          "name": "Salade Niçoise",
          "seasonings": Array [
            "Unrefined salt",
            "black pepper",
          ],
          "steps": Array [
            "Drain tuna, chop lettuce then halve eggs and olives.",
            "Mix ingredients together then drizzle with lemon juice and olive oil. Season with italian herbs, salt and pepper as needed.",
          ],
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <ActionButton
    active={false}
    activeOpacity={0.85}
    autoInactive={true}
    backdrop={false}
    backgroundTappable={false}
    bgColor="transparent"
    bgOpacity={1}
    buttonColor="#FF5722"
    buttonText="+"
    buttonTextStyle={Object {}}
    degrees={45}
    fixNativeFeedbackRadius={false}
    hideShadow={true}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    offsetX={30}
    offsetY={30}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOn={[Function]}
    outRangeScale={1}
    position="right"
    resetToken={null}
    size={56}
    spacing={20}
    testID="addButton"
    useNativeFeedback={true}
    verticalOrientation="up"
  />
</KeyboardAvoidingView>
`;

exports[`RecipesList component renders as expected 2`] = `
Array [
  Array [
    Object {
      "navBarBackgroundColor": "#81b640",
      "navBarButtonColor": "#ffff",
      "navBarComponentAlignment": "center",
      "navBarCustomView": "recipeBrowser.RecipesListTopBar",
      "navBarCustomViewInitialProps": Object {},
      "navBarTextColor": "#ffff",
      "navBarTextFontFamily": "Avenir",
      "statusBarColor": "#508607",
      "statusBarTextColorScheme": "light",
    },
  ],
]
`;

exports[`RecipesList component renders as expected on android 1`] = `
<KeyboardAvoidingView
  keyboardVerticalOffset={60}
  style={
    Object {
      "flex": 1,
    }
  }
  testID="recipesListScreen"
>
  <FlatList
    contentContainerStyle={
      Object {
        "paddingBottom": 10,
      }
    }
    data={
      Array [
        Object {
          "description": "This buffet classic can also be healthy! Mayonnaise is replaced by cottage cheese. You can blend the filling for more creaminess.",
          "id": 9,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/mustard_deviled_eggs.jpg",
          "ingredients": Array [
            Object {
              "amount": 4,
              "name": "hard-boiled eggs",
              "unit": "",
            },
            Object {
              "amount": 2,
              "name": "Dijon mustard",
              "unit": "tsp",
            },
            Object {
              "amount": 100,
              "name": "cottage cheese",
              "unit": "g",
            },
            Object {
              "amount": 2,
              "name": "chopped dill",
              "unit": "tbsp",
            },
          ],
          "ingredientsNames": "hard-boiled eggs;dijon mustard;cottage cheese;chopped dill;",
          "name": "Mustard Deviled Eggs",
          "seasonings": Array [
            "Paprika powder",
          ],
          "steps": Array [
            "Halve eggs and remove yolk.",
            "Mix yolks with mustard, cottage cheese and half of dill.",
            "Fill egg halves with yolk mixture and garnish with remaining dill.",
          ],
        },
        Object {
          "description": "Scrambled eggs are always a great idea. This version with spinach, tomatoes and salty feta is suitable at any point during the day.",
          "id": 2,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/feta_scrambled_eggs.jpg",
          "ingredients": Array [
            Object {
              "amount": 3,
              "name": "eggs",
              "unit": "",
            },
            Object {
              "amount": 100,
              "name": "low-fat feta cheese",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "diced tomato",
              "unit": "",
            },
            Object {
              "amount": 40,
              "name": "fresh spinach",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "coconut oil",
              "unit": "tsp",
            },
          ],
          "ingredientsNames": "eggs;low-fat feta cheese;diced tomato;fresh spinach;coconut oil;",
          "name": "Feta Scrambled Eggs",
          "seasonings": Array [
            "Grated nutmeg",
            "unrefined salt",
            "black pepper",
          ],
          "steps": Array [
            "Heat up coconut oil in a pan. Cook tomatoes and spinach on medium heat for 2 minutes or until spinach softens.",
            "Crack eggs into pan and cook for approx. 4 minutes or desired consistency.",
            "Add grated nutmeg, salt and pepper.",
            "Crumble feta cheese over warm eggs.",
          ],
        },
        Object {
          "description": "This fruity combination of grapefruit, jalapeños and avocado will invite memories of holidays. A side of quickly fried fish and red cabbage.",
          "id": 31,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/fried_fish_with_grapefruit_salsa.jpg",
          "ingredients": Array [
            Object {
              "amount": 200,
              "name": "white fish fillet (cod, plaice, sole; fresh or frozen)",
              "unit": "g",
            },
            Object {
              "amount": 0.5,
              "name": "avocados",
              "unit": "",
            },
            Object {
              "amount": 0.5,
              "name": "grapefruits",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "chopped jalapeño",
              "unit": "",
            },
            Object {
              "amount": 0.5,
              "name": "coriander",
              "unit": "bunches",
            },
            Object {
              "amount": 0.5,
              "name": "limes (juice)",
              "unit": "",
            },
            Object {
              "amount": 100,
              "name": "chopped cabbage",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "coconut oil",
              "unit": "tsp",
            },
          ],
          "ingredientsNames": "white fish fillet (cod, plaice, sole; fresh or frozen);avocados;grapefruits;chopped jalapeño;coriander;limes (juice);chopped cabbage;coconut oil;",
          "name": "Fried Fish With Grapefruit Salsa",
          "seasonings": Array [
            "Chili flakes",
            "unrefined salt",
            "black pepper",
          ],
          "steps": Array [
            "Peel and cube avocado and grapefruit.",
            "In a bowl, mix avocado, grapefruit, jalapeño, coriander and lime juice. Add a pinch of salt.",
            "Heat up coconut oil in a pan. Fry each side of fish fillets for 3 minutes on medium heat.",
            "Serve fried fish on a bed of chopped red cabbage and grapefruit salsa.",
          ],
        },
        Object {
          "description": "A timeless dish, that has a lot to offer beyond its taste! Why not make this restaurant classic on your own?",
          "id": 61,
          "image": "https://nutrition-assets.freeletics.com/recipes/large/classic_tuna_salad.jpg",
          "ingredients": Array [
            Object {
              "amount": 150,
              "name": "water-packed tuna",
              "unit": "g",
            },
            Object {
              "amount": 2,
              "name": "hard-boiled eggs",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "sliced onion",
              "unit": "",
            },
            Object {
              "amount": 50,
              "name": "pitted green olives",
              "unit": "g",
            },
            Object {
              "amount": 1,
              "name": "diced tomato",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "small romaine lettuce",
              "unit": "",
            },
            Object {
              "amount": 0.5,
              "name": "lemons (juice)",
              "unit": "",
            },
            Object {
              "amount": 1,
              "name": "virgin olive oil",
              "unit": "tsp",
            },
            Object {
              "amount": 1,
              "name": "mixed Italian herbs",
              "unit": "tsp",
            },
          ],
          "ingredientsNames": "water-packed tuna;hard-boiled eggs;sliced onion;pitted green olives;diced tomato;small romaine lettuce;lemons (juice);virgin olive oil;mixed italian herbs;",
          "name": "Salade Niçoise",
          "seasonings": Array [
            "Unrefined salt",
            "black pepper",
          ],
          "steps": Array [
            "Drain tuna, chop lettuce then halve eggs and olives.",
            "Mix ingredients together then drizzle with lemon juice and olive oil. Season with italian herbs, salt and pepper as needed.",
          ],
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <ActionButton
    active={false}
    activeOpacity={0.85}
    autoInactive={true}
    backdrop={false}
    backgroundTappable={false}
    bgColor="transparent"
    bgOpacity={1}
    buttonColor="#FF5722"
    buttonText="+"
    buttonTextStyle={Object {}}
    degrees={45}
    fixNativeFeedbackRadius={false}
    hideShadow={true}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    offsetX={30}
    offsetY={30}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOn={[Function]}
    outRangeScale={1}
    position="right"
    resetToken={null}
    size={56}
    spacing={20}
    testID="addButton"
    useNativeFeedback={true}
    verticalOrientation="up"
  />
</KeyboardAvoidingView>
`;

exports[`RecipesList component renders recipe flat list item as expected 1`] = `
<RecipeCard
  onPress={[Function]}
  recipe={
    Object {
      "description": "This buffet classic can also be healthy! Mayonnaise is replaced by cottage cheese. You can blend the filling for more creaminess.",
      "id": 9,
      "image": "https://nutrition-assets.freeletics.com/recipes/large/mustard_deviled_eggs.jpg",
      "ingredients": Array [
        Object {
          "amount": 4,
          "name": "hard-boiled eggs",
          "unit": "",
        },
        Object {
          "amount": 2,
          "name": "Dijon mustard",
          "unit": "tsp",
        },
        Object {
          "amount": 100,
          "name": "cottage cheese",
          "unit": "g",
        },
        Object {
          "amount": 2,
          "name": "chopped dill",
          "unit": "tbsp",
        },
      ],
      "name": "Mustard Deviled Eggs",
      "seasonings": Array [
        "Paprika powder",
      ],
      "steps": Array [
        "Halve eggs and remove yolk.",
        "Mix yolks with mustard, cottage cheese and half of dill.",
        "Fill egg halves with yolk mixture and garnish with remaining dill.",
      ],
    }
  }
  testID="recipeCard_9"
/>
`;
